### ISSUE TITLE
Timeout while downloading Artifacts

### ISSUE THREAD

**Issue by ronnykwon** [microsoft/navcontainerhelper#3555]
**BCContainerHelper 6.0.17 timeouts when downloading artifacts**
Trying to download artifacts from both bcinsider and bcartifacts storage.
After 5 mns of downloading, the bccontainerhelper would raise an error : 
```
Downloading artifact /sandbox/24.1.18927.20043/fr
Downloading C:\Users\ronny\AppData\Local\Temp\4d47bdec-9358-413c-9456-c5348d593993.zip
Could not download from CDN..., retrying from blob storage in 6 seconds...
```
**Scripts used to create container and cause the issue**
```
$artifact = Get-BCArtifactUrl -storage bcartifacts -type sandbox -version 24.1 -accept_insiderEula -country fr -select latest
Download-Artifacts $artifact
```

**Full output of scripts**
```
PS C:\Windows\System32> Import-module bccontainerhelper
BcContainerHelper version 6.0.17
WinRM service is not running, will not try to use WinRM sessions
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Running on Windows, PowerShell 7.4.2
WARNING: The names of some imported commands from the module 'bccontainerhelper' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
PS C:\Windows\System32> $artifact = Get-BCArtifactUrl -storage bcartifacts -type sandbox -version 24.1 -accept_insiderEula -country fr -select latest
PS C:\Windows\System32> Download-Artifacts $artifact
Downloading artifact /sandbox/24.1.18927.20043/fr
Downloading C:\Users\ronny\AppData\Local\Temp\4d47bdec-9358-413c-9456-c5348d593993.zip
Could not download from CDN..., retrying from blob storage in 6 seconds...
```

**Additional context**
- it happens very very frequently
- I've noticed it started after the release of BC 24. I've never had this issue before.

**Comment by freddydk** [comment]
Could you try to run:

```
$bcContainerHelperConfig.artifactDownloadTimeout = 0
```

**Comment by ronnykwon** [comment]
I have this error 
```
PS C:\Windows\System32> $bcContainerHelperConfig.artifactDownloadTimeout = 0
PS C:\Windows\System32> Download-Artifacts $artifact
Downloading artifact /sandbox/24.1.18927.20043/fr
Downloading C:\Users\ronny\AppData\Local\Temp\7185dd16-8833-49c4-83be-a6a3bbb133e4.zip
Could not download from CDN..., retrying from blob storage in 5 seconds...
Download-Artifacts Telemetry Correlation Id: c2e39fcd-85c0-45a6-96fd-97d9f72e99ae
Exception: C:\Users\ronny\OneDrive - Capset\Documents\PowerShell\Modules\bccontainerhelper\6.0.17\Common\Download-File.ps1:95
Line |
  95 |                  throw (GetExtendedErrorMessage $_)
     |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception setting "Timeout": "value ('00:00:00') must be greater than '00:00:00'. (Parameter 'value') Actual value was 00:00:00."
```

**Comment by freddydk** [comment]
The artifactDownloadTimeout is default 300 seconds - but I will increase that to 600 seconds.
If your runners are slower than this, you would need to add this setting to bccontainerhelper.settings.json on your runners (or specify it in al-go repo settings if running AL-Go for GitHub)

**Comment by ronnykwon** [comment]
which settings exactly in the al-go repo settings ? I can't find it on https://github.com/microsoft/AL-Go/blob/main/Scenarios/settings.md

**Comment by freddydk** [comment]
in .github/AL-Go-Settings.json you can specify settings for BcContainerHelper directly.

**Comment by ronnykwon** [comment]
So I added the config in the AL-GO-Settings.json but It's skipped during the build phase : 
```
Import from C:\ProgramData\BcContainerHelper\6.0.17\BcContainerHelper\BcContainerHelper.ps1
BcContainerHelper version 6.0.17
Ignoring config setting artifactDownloadTimeout as the type in the config file is different than in the default configuration
Ignoring config setting defaultDownloadTimeout as the type in the config file is different than in the default configuration
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Running on Windows, PowerShell 7.4.2
use settings and secrets
Analyzing repository
  Checking type
  Checking appFolders, testFolders and bcptTestFolders
  Application Dependency 24.0.0.0
  Checking artifact setting for project
  Downloading artifacts from https://bcartifacts-exdbf9fwegejdqak.b02.azurefd.net/sandbox/24.0.16410.20025/fr
  Downloading platform artifact /sandbox/24.0.16410.20025/platform
  Downloading C:\Users\Administrateur\AppData\Local\Temp\66f7a408-59c8-45d4-8f1e-652ee873[68](https://github.com/CapsetPF/Capset_AL_Framework/actions/runs/9260111950/job/25473310589#step:8:70)ae.zip
  Could not download from CDN..., retrying from blob storage in 2 seconds...
```

**Comment by freddydk** [comment]
That is actually a new bug caused by a type difference in PowerShell 7
Only thing you can do right now is to run PowerShell 5.1
I will fix the issue in ContainerHelper asap

**Comment by freddydk** [comment]
Fixed in 6.0.18
